### ISSUE TITLE
lightmapper_cpu.cpp: 'rand' not declared in this scope when compiling

### ISSUE THREAD

**Issue by ChainedLupine** [godotengine/godot#45342]
**Godot version:**
3.2.4 beta 6 from 7e207cfd48d6077ac6aaa3c45423d3fcf2f90bd7

**OS/device including version:**
Windows 10 (building under WSL, Ubuntu 16.04.7 LTS)

**Issue description:**
Expect completed build, but get:
```
modules/lightmapper_cpu/lightmapper_cpu.cpp: In function 'float uniform_rand()':
modules/lightmapper_cpu/lightmapper_cpu.cpp:779:44: error: 'rand' was not declared in this scope
  static thread_local uint32_t state = rand();
                                            ^
```

**Steps to reproduce:**
1. Build with `scons -j32 platform=x11 use_lto=yes bits=64 tools=yes target=release_debug`

**Note: tools=no does not generate this error.**

**Minimal reproduction project:**
N/A

**Comment by ChainedLupine** [comment]
Inserting `#include <random>` into lightmapper_cpu.h fixes it but I don't know if this causes problems with other platforms.

**Comment by akien-mga** [comment]
No compilation issue here but maybe older GCC versions had different base includes. I'm assuming you're using gcc-5.3.1 from Ubuntu 16.04?

**Comment by akien-mga** [comment]
Does this change work? It should be more minimal than including `<random>`:
```diff
diff --git a/modules/lightmapper_cpu/lightmapper_cpu.cpp b/modules/lightmapper_cpu/lightmapper_cpu.cpp
index ba07451cdf..75af65ed90 100644
--- a/modules/lightmapper_cpu/lightmapper_cpu.cpp
+++ b/modules/lightmapper_cpu/lightmapper_cpu.cpp
@@ -36,6 +36,8 @@
 #include "core/project_settings.h"
 #include "modules/raycast/lightmap_raycaster.h"
 
+#include <cstdlib> // rand
+
 Error LightmapperCPU::_layout_atlas(int p_max_size, Vector2i *r_atlas_size, int *r_atlas_slices) {
 
        Vector2i atlas_size;
```

**Comment by asmaloney** [comment]
Just ran into this using latest clang on macOS as well.

The fix I was about to make a PR for was to use `Math`:

```cpp
static thread_local uint32_t state = Math::rand();
```

**Comment by ChainedLupine** [comment]
> No compilation issue here with GCC 10 but maybe older GCC versions had different base includes. I'm assuming you're using gcc-5.3.1 from Ubuntu 16.04?

Current versions:

* gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.12)
* ldd (Ubuntu GLIBC 2.23-0ubuntu11.2) 2.23
* GNU C Library (Ubuntu GLIBC 2.23-0ubuntu11.2) stable release version 2.23, by Roland McGrath et al.

> As a side note, I would advise that you install the newest GCC or Clang you can get on Ubuntu 16.04 to have better optimization and LTO. GCC 5 is really old by now and there's been a lot of improvement over the years in compilers. (On the other hand you're right about building on Ubuntu 16.04 because you need an old glibc for compatibility with all distros with a newer version).

Good point about updating GCC!  And yeah, the 16.04 is my original WSL version from ages ago and it comes in handy at times. :)

**Comment by ChainedLupine** [comment]
I tested both @akien-mga and @asmaloney proposed fixes and both worked.

**Comment by akien-mga** [comment]
Fixed by #45353.
